src/main/java/com/selfxdsd/selfweb/api/WalletsApi.java

```diff
@@ -189,10 +192,11 @@ public WalletsApi(final User user, final Self self) {
                     response = ResponseEntity.badRequest()
                         .body("Wallet of type " + type + " not found.");
                 } else {
+                    final BigDecimal cash = BigDecimal.valueOf(limit)
+                        .setScale(2, RoundingMode.HALF_UP)
```

Comment:
@amihaiemilI've rounded limit to up to 2 decimals, this is reflected in tests too.